### DISCUSSION TITLE
Local LAN API for Gateway

### DISCUSSION THREAD

**Discussion by apwelsh** [iopool/community#12] [Ideas]
Thank you for the public API.  This will enable integrators to start creating 3rd party support for IOPool EcO.  The next step, is a local API hosted directly on the gateway.  This will enable integrators to talk directly to the gateway, and not have to rely on cloud services to poll the devices serviced by the gateway.  API Keys could be created by using the IOPool app to request the key from the gateway device, and then all data could then be polled locally, and securely, since access would only be granted if a valid API key is provided.

**Comment by mehdyouras** [comment]
Thanks for the suggestion. Right now it's not on our roadmap but i'll be pleased to add it to our feature requests list.

**Comment by jimtng** [comment]
Any progress on the local API?

**Comment by KLucky13** [comment]
This point became even stronger, cloud API stopped working since last week. Can't get it to work anymore. A local API would be much better!

**Reply by Jeoffreybauvin** [reply]
@KLucky13  What are you talking about ? Cloud API is working fine, I'm using right now : 

![image](https://github.com/user-attachments/assets/49b12a47-79d4-4737-b5c6-b3189b55a380)

**Reply by KLucky13** [reply]
For me it stopped working since last week.. => https://github.com/iopool/community/discussions/23
